src/encoder.rs

```diff
@@ -1337,50 +1383,9 @@ pub fn encode_block_b(seq: &Sequence, fi: &FrameInvariants, fs: &mut FrameState,
         cw.write_use_filter_intra(w,false, bsize); // Always turn off FILTER_INTRA
     }
 
-    if is_inter {
-      {
-        let ref_frame = cw.bc.at(bo).ref_frames[0];
-        let mv = &cw.bc.at(bo).mv[0];
-        // Inter mode prediction can take place once for a whole partition,
-        // instead of each tx-block.
-        let num_planes = 1 + if has_chroma(bo, bsize, xdec, ydec) { 2 } else { 0 };
-        for p in 0..num_planes {
-            let plane_bsize = if p == 0 { bsize }
-            else { get_plane_block_size(bsize, xdec, ydec) };
+    motion_compensate(fi, fs, cw, luma_mode, ref_frame, mv, bsize, bo, bit_depth);
```

Comment:
Shouldn't this be called only for inter mode block? i.e. inside the below if is_inter {...} ?